src/components/__tests__/useItemReverseIndex.test.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom/client';
+import { IMenuItem } from '../..';
+import useItemReverseIndex, {
+  MenuItemReverseIndexToVirtualIndex,
+} from '../Virtualization/useItemReverseIndex';
+
+(global as any).IS_REACT_ACT_ENVIRONMENT = true;
+
+describe('useItemReverseIndex', () => {
+  let container: HTMLDivElement;
+  let root: ReactDOM.Root;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+    root = ReactDOM.createRoot(container);
+  });
+
+  afterEach(() => {
+    React.act(() => root.unmount());
+    document.body.removeChild(container);
+    container = null as any;
+  });
+
+  it('it should maps item based on the id for quick lookup', () => {
+    const menuItems: any[] = [
+      { id: 'item1', name: 'Item 1' },
+      { id: 'item2', name: 'Item 2' },
+      { id: 'item3', name: 'Item 3' },
+    ];
+
+    let result: MenuItemReverseIndexToVirtualIndex | undefined;
+
+    // Hooks can only be used inside a React component.
+    // This component is just to host the hook.
+    const TestComponent = ({ items }: { items: IMenuItem[] }) => {
```

Comment:
minor ,  the TestComponent shows up a lot in the tests. Could be a spot to apply DRY , but I leave it up to you to decide